apis/network/v1alpha3/zone_types.go

```diff
@@ -0,0 +1,133 @@
+
+	"github.com/aws/aws-sdk-go-v2/service/route53"
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// +kubebuilder:object:root=true
+
+// Zone is a managed resource that represents an AWS Route53 Hosted Zone.
+// +kubebuilder:printcolumn:name="ID",type="string",JSONPath=".status.atProvider.Id"
+// +kubebuilder:printcolumn:name="RRs",type="integer",JSONPath=".status.atProvider.ResourceRecordSetCount"
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type Zone struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ZoneSpec   `json:"spec"`
+	Status ZoneStatus `json:"status,omitempty"`
+}
+
+// ZoneSpec defines the desired state of an AWS Route53 Hosted Zone.
+type ZoneSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ZoneParameters `json:"forProvider"`
+}
+
+// ZoneStatus represents the observed state of a Zone.
+type ZoneStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     ZoneObservation `json:"atProvider"`
+}
+
+// ZoneParameters define the desired state of an AWS Route53 Hosted Zone.
+type ZoneParameters struct {
+	// CallerReference is an unique string that identifies the request and that
+	// allows failed Zone create requests to be retried without the risk of
+	// executing the operation twice.
+	// +immutable
+	CallerReference *string `json:"callerref"`
+
+	// Any comments that you want to include about the hosted zone.
+	Comment *string `json:"comment,omitempty"`
+
+	// A value that indicates whether this is a private hosted zone.
+	// +immutable
+	PrivateZone *bool `json:"privatezone,omitempty"`
+
+	// The name of the domain. Specify a fully qualified domain name, for example,
+	// www.example.com. The trailing dot is optional; Amazon Route 53 assumes that
+	// the domain name is fully qualified. This means that Route 53 treats www.example.com
+	// (without a trailing dot) and www.example.com. (with a trailing dot) as identical.
+	//
+	// If you're creating a public hosted zone, this is the name you have registered
+	// with your DNS registrar. If your domain name is registered with a registrar
+	// other than Route 53, change the name servers for your domain to the set of
+	// NameServers that CreateHostedZone returns in DelegationSet.
+	// +immutable
+	Name *string `json:"name"`
+
+	// (Private hosted zones only) The ID of an Amazon VPC that you're
+	// associating with this hosted zone. You can specify only one Amazon VPC
+	// when you create a private hosted zone.
+	// +immutable
+	VPCId *string `json:"VPCId,omitempty"`
```

Comment:
It seems like this is a reference to a VPC. Should it have a corresponding `VPCIdRef` and `VPCIdSelector`?

Reply:
At the moment when this PR was raised the VPC referencer wasn't implemented but has been implemented now.